RadarSDK/RadarFileStorage.m

```diff
@@ -0,0 +1,58 @@
+//
+//  RadarFileStorage.m
+//  RadarSDK
+//
+//  Created by Kenny Hu on 12/6/23.
+//  Copyright © 2023 Radar Labs, Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "RadarFileStorage.h"
+
+@implementation RadarFileStorage
+
+
+
+- (NSData *)readFileAtPath:(NSString *)filePath {
+    __block NSData *fileData = nil;
+
+    NSFileCoordinator *fileCoordinator = [[NSFileCoordinator alloc] init];
+    [fileCoordinator coordinateReadingItemAtURL:[NSURL fileURLWithPath:filePath] options:0 error:nil byAccessor:^(NSURL *newURL) {
+        fileData = [NSData dataWithContentsOfURL:newURL];
+    }];
+
+    return fileData;
+}
+
+- (void)writeData:(NSData *)data toFileAtPath:(NSString *)filePath {
+
+    NSFileCoordinator *fileCoordinator = [[NSFileCoordinator alloc] init];
+    [fileCoordinator coordinateWritingItemAtURL:[NSURL fileURLWithPath:filePath] options:NSFileCoordinatorWritingForReplacing error:nil byAccessor:^(NSURL *newURL) {
+        [data writeToURL:newURL options:NSDataWritingAtomic error:nil];
+    }];
+}
+
+- (void)deleteFileAtPath:(NSString *)filePath {
+    NSFileCoordinator *fileCoordinator = [[NSFileCoordinator alloc] init];
+    [fileCoordinator coordinateWritingItemAtURL:[NSURL fileURLWithPath:filePath] options:NSFileCoordinatorWritingForDeleting error:nil byAccessor:^(NSURL *newURL) {
+        [[NSFileManager defaultManager] removeItemAtURL:newURL error:nil];
+    }];
+}
+
+- (NSArray<NSString *> *)allFilesInDirectory:(NSString *)directoryPath {
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    NSError *error = nil;
+    NSArray<NSString *> *files = [fileManager contentsOfDirectoryAtPath:directoryPath error:&error];
+    
+    if (error) {
+        NSLog(@"Failed to get files in directory: %@", [error localizedDescription]);
+        return nil;
+    }
+    
+    return [files sortedArrayUsingComparator:^NSComparisonResult(NSString *str1, NSString *str2) {
+        return [@([str1 integerValue]) compare:@([str2 integerValue])];
+    }];
```

Comment:
This code only sorts correctly IFF file names are unix timestamps.

Thinking what to do here...

Reply:
We definitely should have a fallback of some sort in the case that they're not

Reply:
Oh yeah, I made this function too specific for logs. This method should use HOFs. I'm thinking I add an another argument for a `NSComparator` and the consumer of the method can define the ordering of the sort.